src/slurm_plugin/slurm_resources.py

```diff
@@ -266,6 +268,11 @@ def _parse_error_code(self):
             return error_code
         return None
 
+    @abstractmethod
+    def is_powered_down_and_not_ice(self):
```

Comment:
same here, shall we have this logic in the caller?

Reply:
Removed the function, in the clustermgtd, check `if not node.is_ice() and node.is_power() and not node.is_nodeaddr_set():`